src/Orleans/Utils/WrappedException.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Runtime.Serialization;
+
+namespace Orleans.Runtime.Internal
+{
+    /// <summary>Exception used to communicate with the storage provider, so that it throws this exception to its caller.</summary>
+    [Serializable]
+    public class WrappedException : OrleansException
```

Comment:
BTW, OrleansProviders does NOT reference OrleansRuntime. Originally I attempted to add this exception as a nested type inside MemoryStorageGrain, but the MemoryStorage did not have access to it. Also, can't create a nested type inside the IMemoryStorageGrain (cannot nest types in interfaces), so my only option is to create a full blown type in Orleans itself... but I'll make it internal now